BraintreeModels/src/main/java/com/braintreepayments/api/PaymentMethodNonce.java

```diff
@@ -2,60 +2,60 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
 
 /**
  * Base class representing a method of payment for a customer. {@link PaymentMethodNonce} represents the
  * common interface of all payment method nonces, and can be handled by a server interchangeably.
  */
-public abstract class PaymentMethodNonce implements Parcelable {
+public class PaymentMethodNonce implements Parcelable {
 
-    private static final String PAYMENT_METHOD_NONCE_COLLECTION_KEY = "paymentMethods";
     private static final String PAYMENT_METHOD_TYPE_KEY = "type";
     private static final String PAYMENT_METHOD_NONCE_KEY = "nonce";
     private static final String PAYMENT_METHOD_DEFAULT_KEY = "default";
     private static final String DESCRIPTION_KEY = "description";
 
-    static final String DATA_KEY = "data";
-    static final String TOKEN_KEY = "token";
+    private final String mNonce;
+    private final String mDescription;
+    private final boolean mDefault;
 
-    protected String mNonce;
-    protected String mDescription;
-    protected boolean mDefault;
+    private @PaymentMethodType final int mType;
 
-    static JSONObject getJsonObjectForType(String apiResourceKey, JSONObject json) throws JSONException {
-        return json.getJSONArray(apiResourceKey).getJSONObject(0);
+    static PaymentMethodNonce fromJSON(JSONObject inputJson) throws JSONException {
+        String typeString = inputJson.getString(PAYMENT_METHOD_TYPE_KEY);
+        int type = paymentMethodTypeFromString(typeString);
+
+        String nonce = inputJson.getString(PAYMENT_METHOD_NONCE_KEY);
+        String description = inputJson.getString(DESCRIPTION_KEY);
+        boolean isDefault = inputJson.optBoolean(PAYMENT_METHOD_DEFAULT_KEY, false);
+
+        return new PaymentMethodNonce(nonce, description, isDefault, type);
     }
 
-    @CallSuper
-    void fromJson(JSONObject json) throws JSONException {
-        mNonce = json.getString(PAYMENT_METHOD_NONCE_KEY);
-        mDescription = json.getString(DESCRIPTION_KEY);
-        mDefault = json.optBoolean(PAYMENT_METHOD_DEFAULT_KEY, false);
+    PaymentMethodNonce(String nonce, String description, boolean isDefault, @PaymentMethodType int type) {
+        mNonce = nonce;
+        mDescription = description;
+        mDefault = isDefault;
+        mType = type;
     }
 
     /**
      * @return The nonce generated for this payment method by the Braintree gateway. The nonce will
      *          represent this PaymentMethod for the purposes of creating transactions and other monetary
      *          actions.
      */
-    public String getNonce() {
+    public String getString() {
         return mNonce;
     }
 
     /**
      * @return The description of this PaymentMethod for displaying to a customer, e.g. 'Visa ending in...'
      */
     public String getDescription() {
+        // TODO: find a way to localize this, or move to Drop-In
```

Comment:
I think we want to just remove this (if we want to, we can add it back in later based on merchant feedback)

Reply:
Removed this 👍